### PR TITLE
Support cross-region replication

### PR DESCRIPTION
**Pull request by Nuru**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
     aws = {
-      source  = "hashicorp/aws"
-      version = ">= 2.0"
-    }
-    local = {
-      source  = "hashicorp/local"
-      version = ">= 1.3"
+      source                = "hashicorp/aws"
+      version               = ">= 4.0"
+      configuration_aliases = [aws.blue, aws.green]
```
[the remark this answers is not in the document]
Comment (Nuru) [reply versions.tf:8]: The new module requires the caller to provide 2 AWS providers, even if they are not using cross-region replication. This is a limitation of Terraform. However, as explained [here](https://github.com/cloudposse/terraform-aws-tfstate-backend/blob/71c8ef6e40986c7331ef53008695a01d2c4651c4/docs/migration-v0-v1.md#easy) and to be explained in the README, you can use the same provider for both. There is an example in `examples/no-replication:  https://github.com/cloudposse/terraform-aws-tfstate-backend/blob/71c8ef6e40986c7331ef53008695a01d2c4651c4/examples/no-replication/main.tf#L8-L11

File: docs/migration-v0-v1.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+target the secondary region.
+- Because of the above, this module now requires 2 AWS providers: `aws.blue` 
+and `aws.green`, configured for the primary and secondary regions respectively.
+If you do not want to enable replication and a second bucket, you still must
+pass in providers for both colors, but you can use the same provider for both.
+- When replication is enabled, an SNS topic is created to publish notifications
+of replication events. You must subscribe to this topic to receive notifications.
+- Server-side encryption is enabled for the S3 buckets and cannot be disabled.
+There is therefore no longer a block against unencrypted uploads, since they
+will be immediately encrypted anyway.
```
Comment (mcalhoun) [inline docs/migration-v0-v1.md:16]: Doesn't this mean we're no longer enforcing the following?:

```
 "Condition": {
        "Bool": {
          "aws:SecureTransport": "false"
        }
      },
```

If so, some of the compliance checks are going to fail. This means that the object isn't required to be encrypted in transit.

**Comment by Gowiem** [comment]
Hey @Nuru @mcalhoun @aknysh is this worth keeping around? Totally fine if it's something ya'll are planning on circling back on, but figured I should raise the questions considering it has drifted so much and is very stale at this point.

**Comment by hans-d** [comment]
Closing due to staleness, feel free to reopen and update into a reviewable state.

**Pull request closed without merging**